pkg/cluster/context.go

```diff
@@ -349,6 +351,17 @@ func (cc *createContext) provisionControlPlane(
 		}
 	}
 
+	// Wait for control plane node to reach Ready status.
+	if cc.waitForControlPlane {
+		if !node.WaitForControlPlane(time.Now().Add(time.Second * 30)) {
```

Comment:
yes, let's make this configurable

Reply:
kubeadm has an initial timeout of 40sec for the kubelet to ready up post init and  in parallel it has a healthz check for the api server that runs for 5 minutes (config via TimeoutForControlPlane in v1beta1).
https://godoc.org/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm#APIServer

Reply:
so the control plane is up, but that doesn't mean non-control plane workloads are ready to schedule yet. I think default should be zero (cluster create is snappy locally), and configurable for CI where you can decide yourself how long is worth waiting before just trying to run your tests.

If your tests fail, you need to identify why already, and this way it's up to you how much you want to try to wait. 

The wait still won't be a strong guarantee, and I don't think we want to solve that yet, that seems like something that should be solved with a kubeadm command instead of being implemented just for `kind`.

On manual local usage, you won't be instantly trying to schedule workloads anyhow.